context/connection.go

```diff
@@ -117,29 +133,38 @@ func FindConnectionByURL(ctx Context, connectionString string) (*models.Connecti
 		return &connection, nil
 	}
 
-	name, namespace, connectionType, found := extractConnectionNameType(connectionString)
+	name, namespace, found := extractConnectionNameType(connectionString)
 	if !found {
 		return nil, nil
 	}
 
-	connection, err := FindConnection(ctx, connectionType, name, namespace)
+	connection, err := FindConnection(ctx, name, namespace)
 	if err != nil {
-		return nil, fmt.Errorf("failed to find connection (type=%s, name=%s, namespace=%s): %w", connectionType, name, namespace, err)
+		return nil, fmt.Errorf("failed to find connection (name=%s, namespace=%s): %w", name, namespace, err)
 	}
 
 	return connection, nil
 }
 
 // FindConnection returns the connection with the given type and name
-func FindConnection(ctx Context, connectionType, name, namespace string) (*models.Connection, error) {
+func FindConnection(ctx Context, name, namespace string) (*models.Connection, error) {
 	var connection models.Connection
 
 	if namespace == "" {
 		namespace = ctx.GetNamespace()
 	}
 
-	err := ctx.DB().Where("type = ? AND name = ? AND namespace = ?", connectionType, name, namespace).First(&connection).Error
-	if err != nil {
+	// NOTE: For backward compatibility reason we use the namespace as the connection type
+	// Before: connection://<type>/<name>
+	// Now: connection://<namespace>/<name.
+	if err := ctx.DB().Where("name = ? AND type = ?", name, namespace).
```

Comment:
Can we make this this the second option, and also log a warning when its found using this?